src/os/bluestore/BlueFS.cc

```diff
@@ -4195,6 +4589,11 @@ void BlueFS::_close_writer(FileWriter *h)
 }
 void BlueFS::close_writer(FileWriter *h)
 {
+  if (h->file->is_new_wal()) {
+    // we force fsync by forcing dirty flag
+    fsync(h, true);
```

Comment:
I think this is very interesting.
Shouldn't we have unconditional `fsync()` ?

Reply:
this case was added to force update of file metadata: `wal_limit` and `wal_size`. If not, in `_replay`(`wal_update_size`) it would nicely recover, but this feels like we leave the file in the correct state always instead of relying on `_replay`